README.md

```diff
@@ -636,6 +653,136 @@ When defining an Audience, users and groups can be excluded from the audience. T
+
+### Assigning a Variant
+
+The process of deciding which variant to return for a feature is called assignment. Assignment is determined by the `Allocation` property of the feature. Allocation logic is similar to the [Microsoft.Targeting](./README.md#MicrosoftTargeting) feature filter, with some additional parameters to allow for more specific variant assignment.
+
+```
+"Allocation": { 
+    "DefaultWhenEnabled": "Small", 
+    "DefaultWhenDisabled": "Small",  
+    "User": [ 
+        { 
+            "Variant": "Big", 
+            "Users": [ 
+                "Marsha" 
+            ] 
+        } 
+    ], 
+    "Group": [ 
+        { 
+            "Variant": "Big", 
+            "Groups": [ 
+                "Ring1" 
+            ] 
+        } 
+    ],
+    "Percentile": [ 
+        { 
+            "Variant": "Big", 
+            "From": 0, 
+            "To": 10 
+        } 
+    ], 
+    "Seed": "13973240" 
+},
+"Variants": [
+    { 
+        "Name": "Big", 
+        "ConfigurationReference": "ShoppingCart:Big" 
+    },  
+    { 
+        "Name": "Small", 
+        "ConfigurationValue": {
+            "Size": 300
+        }
+    } 
+]
+```
+
+In the above example, if the feature is not enabled, `GetVariantAsync` would return the variant allocated by `DefaultWhenDisabled`, which is `Small` in this case. 
+
+If the feature is enabled, the feature manager will check the `User`, `Group`, and `Percentile` allocations in that order to see if they match the targeting context or calculated percentile for that context. If the user is named `Marsha`, in the group named `Ring1`, or the user happens to fall between the 0 and 10th percentile calculated with the given `Seed`, then the specified variant is returned for that allocation. In this case, all of these would return the `Big` variant. If none of these allocations match the targeting context, the `DefaultWhenEnabled` variant is returned.
+
+### StatusOverride
+
+The `StatusOverride` property, if set for the assigned variant, overrides a feature's state. By default, this property is equal to `None`, which does not affect the feature state. If `StatusOverride` is equal to `Enabled` for the assigned variant, then the feature is enabled. If `StatusOverride` is equal to `Disabled` for the assigned variant, then the feature is disabled.
+
+```
+"Allocation": { 
+    "DefaultWhenEnabled": "OffVariant"
+},
+"Variants": [
+    { 
+        "Name": "OffVariant", 
+        "ConfigurationValue": {
+            "Size": 300
+        },
+        "StatusOverride": "Disabled"
+    }
+],
+"EnabledFor": [ 
+    { 
+        "Name": "On" 
+    } 
+] 
+```
+
+In the above example, the feature is enabled by the `On` filter and assigns the variant set for `DefaultWhenEnabled`, which is the `OffVariant` variant. The `OffVariant` variant has the `StatusOverride` property set to `Disabled`, so calling `IsEnabledAsync` for this feature will return disabled, even though the feature would otherwise be enabled by its filters.

```

Comment:
While nothing is wrong in this example, it's unclear what the `StatusOverride` is for. Can we have a more real-world example?

Reply:
If we do add an example for StatusOverride, we should make it very clear that it's a bandaid for a specific situation. 

That situation is if you want to use allocation to control true and false without making code changes.

If you're writing a new app or adding new features, use variants with allocation. If you want simple true and false, you can achieve it with a targeting filter. If you want "variants" for your existing true and false, use the defaults.

Honestly, I prefer not having an example for StatusOverride as I don't want new customers grabbing the snippet.

Reply:
There must be some misunderstanding. I can agree the `StatusOverride` is a bandaid fix (in terms of design) but it is NOT just for existing users who don't want to make code changes.

The allocation is more powerful than the targeting filter (see an example later). We will want our customers (existing or new) to adopt the allocation with variants. However, without `StatusOverride`, they will have to write everything like 

```cs
if (await fm.GetVariantAsync("foo").Configuration["ConfigurationValue"] == "true") ... 
// or
if (await fm.GetVariantAsync("foo").Name == "On") ... 
```
instead of simply 
```cs
if (await fm.IsEnabledAsync("foo"))...
```
Our whole `MS.FM.AspNetCore` package is essentially built on top of the `IsEnabledAsync` API. Without `StatusOverride`, that package can be thrown out of the window. We all know On/Off accounts for the majority of the real-world usage. This is why `StatusOverride` is important for both existing and new customers.

We should explain what the `StatusOverride` is for and why it's useful. Here is an example that you can't achieve with targeting.

```json
"Allocation": {
    "Percentile": {
        "Variant": "On",
        "From": 10,
        "To": 20
    },
    "DefaultWhenEnabled":  "Off",
    "Seed": "Back-friday-feature-group"
},
"Variants": [
    { 
        "Name": "On", 
        "ConfigurationValue": true,
        "StatusOverride": "Enabled"
    },
    { 
        "Name": "Off", 
        "ConfigurationValue": false,
        "StatusOverride": "Disabled"
    }    
],
"EnabledFor": [ 
    { 
        "Name": "AlwaysOn" 
    } 
] 
```

Reply:
I've pretty much added this example to the StatusOverride section now